server/server.go

```diff
@@ -287,7 +287,8 @@ func newServer(ctx context.Context, c Config, rotationStrategy rotationStrategy)
 		var handler http.Handler = h
 		if len(c.AllowedOrigins) > 0 {
 			corsOption := handlers.AllowedOrigins(c.AllowedOrigins)
-			handler = handlers.CORS(corsOption)(handler)
+			corsHeaders := handlers.AllowedHeaders([]string{"Authorization"})
```

Comment:
Is this change related to PKCE? If not, what's the reason for adding it?

Also, I don't think it's enough if you need to send credentials in a cors request. I think you also need `AllowCredentials` as well.

But if it's not related to the PR and it's there for testing reasons, it should probably be removed.

Reply:
I added `"Authorization"` to the allowed headers that, from a browser, after a successful login with PKCE, the `/userinfo` endpoint can be called with the bearer token from the browser.
I tested it with oidc-client, which has a parameter `loadUserInfo`, which does just that successfully.

According to [this](https://www.sjoerdlangkemper.nl/2018/09/12/authorization-header-and-cors/), whitelisting `"Authorization"` in the `Access-Control-Allow-Headers` is sufficient, if we just want to send a bearer token (and do not want the browser to ask the user for a password).

But it is not essential for PKCE, i would consider it a bug. There is already an issue [1532](https://github.com/dexidp/dex/issues/1532) for it.

I will remove it the.

Reply:
A similar patch is handled in PR #1819

Reply:
Done (reverted).